port/unix/omrsysinfo.c

```diff
@@ -6495,140 +6490,157 @@ readCgroupSubsystemFile(struct OMRPortLibrary *portLibrary, uint64_t subsystemFl
+			 * For any other cgroup name, assume we are in a container.
+			 * Note that this will not work if namespaces are enabled in the container as all
+			 * cgroup names will be the root cgroup.
+			 */
+			if (NULL == (cgroupFile = fopen(OMR_PROC_PID_ONE_CGROUP_FILE, "r"))) {
 				int32_t osErrCode = errno;
-				Trc_PRT_isRunningInContainer_fgets_failed(OMR_PROC_PID_ONE_CGROUP_FILE, osErrCode);
-				rc = portLibrary->error_set_last_error_with_message_format(portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_READ_FAILED, "fgets failed to read %s file stream with errno=%d", OMR_PROC_PID_ONE_CGROUP_FILE, osErrCode);
+				Trc_PRT_isRunningInContainer_fopen_failed(OMR_PROC_PID_ONE_CGROUP_FILE, osErrCode);
+				rc = portLibrary->error_set_last_error_with_message_format(
+						portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_FOPEN_FAILED,
+						"fopen failed to open %s file with errno=%d", OMR_PROC_PID_ONE_CGROUP_FILE, osErrCode);
 				goto _end;
 			}
-			rc = sscanf(buffer, PROC_PID_CGROUPV1_ENTRY_FORMAT, &hierId, subsystems, cgroup);
 
-			if (EOF == rc) {
-				break;
-			} else if (1 == rc) {
-				rc = sscanf(buffer, PROC_PID_CGROUP_SYSTEMD_ENTRY_FORMAT, &hierId, cgroup);
+			while (0 == feof(cgroupFile)) {
+				char buffer[PATH_MAX];
+				char cgroup[PATH_MAX];
+				char subsystems[PATH_MAX];
+				int32_t hierId = -1;
 
-				if (2 != rc) {
+				if (NULL == fgets(buffer, sizeof(buffer), cgroupFile)) {
+					break;
+				}
+				if (0 != ferror(cgroupFile)) {
+					int32_t osErrCode = errno;
+					Trc_PRT_isRunningInContainer_fgets_failed(OMR_PROC_PID_ONE_CGROUP_FILE, osErrCode);
+					rc = portLibrary->error_set_last_error_with_message_format(
+							portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_READ_FAILED,
+							"fgets failed to read %s file stream with errno=%d", OMR_PROC_PID_ONE_CGROUP_FILE, osErrCode);
+					goto _end;
+				}
+				rc = sscanf(buffer, PROC_PID_CGROUPV1_ENTRY_FORMAT, &hierId, subsystems, cgroup);
+
+				if (EOF == rc) {
+					break;
+				} else if (1 == rc) {
+					rc = sscanf(buffer, PROC_PID_CGROUP_SYSTEMD_ENTRY_FORMAT, &hierId, cgroup);
+
+					if (2 != rc) {
+						Trc_PRT_isRunningInContainer_unexpected_format(OMR_PROC_PID_ONE_CGROUP_FILE);
+						rc = portLibrary->error_set_last_error_with_message_format(
+								portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_READ_FAILED,
+								"unexpected format of %s file", OMR_PROC_PID_ONE_CGROUP_FILE);
+						goto _end;
+					}
+				} else if (3 != rc) {
 					Trc_PRT_isRunningInContainer_unexpected_format(OMR_PROC_PID_ONE_CGROUP_FILE);
-					rc = portLibrary->error_set_last_error_with_message_format(portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_READ_FAILED, "unexpected format of %s file", OMR_PROC_PID_ONE_CGROUP_FILE);
+					rc = portLibrary->error_set_last_error_with_message_format(
+							portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_READ_FAILED,
+							"unexpected format of %s file", OMR_PROC_PID_ONE_CGROUP_FILE);
```

Comment:
This has the same message as `(2 != rc)`, it could be different to distinguish the cases.

Reply:
Added more details in those messages to distinguish the cases.